### ISSUE TITLE
Surface.ByPatch Fails on Complex Imported NURBS Curves

### ISSUE THREAD

**Issue by ralvarez76** [DynamoDS/Dynamo#4636]
I am trying to recreate some surfaces that were created in CATIA in Revit. I can import the CATIA file into AutoCAD and get it into Revit, but the number of control points wreaks havoc on the wall by face operations and causes them to fail.

So, instead of working with the original surfaces, I am trying to recreate them in a more simplified way. The loss of fidelity is acceptable in this case because its purpose is to make drawings in Revit.

I am taking the edge curves (loops) of each surface and exporting them to individual AutoCAD files. I then link them into Revit and bring them into Dynamo. In Dynamo I break the curves down into points, rebuild the NURBS and then patch the result into new surfaces.

Out of 65 surfaces I have 5 that refuse to work with Surface.ByPatch. All of the curves are closed polycurves. It gives me the following error:

Warning: Surface.ByPatch operation failed. 
Could not make patch surfaces : API_FAILED -- Operation unsuccessful.

The files are as follows:

![lvmh test edges rfa](https://cloud.githubusercontent.com/assets/2349069/7992966/cde0913a-0ab7-11e5-9907-f65c9a90f975.png)
![edges to patch dyn](https://cloud.githubusercontent.com/assets/2349069/7992965/cddfd542-0ab7-11e5-89c5-c97adb789a34.png)

Rafael Alvarez
Thanks.

**Comment by monikaprabhu** [comment]
@ralvarez76 - could you please email me the .rfa file - I 'm not able to access the file.monika.prabhu@autodesk.com

**Comment by ralvarez76** [comment]
I'm not at a computer right now but the RFA is attached to the message, just rename the extension from PNG to RFA. 

Regards,

Rafael Alvarez

**Comment by ralvarez76** [comment]
@monikaprabhu I emailed you the files. Let me know what you find out. Thanks.

**Comment by TD123** [comment]
@ralvarez76 @monikaprabhu .. did you ever manage to sort this out im having the same issue doing a similar process?

my workflow is project model lines onto a surface > divide points @ X Distance and then use them points to patch a surface however it only works on about 9 out of 20 surfaces.. any help would be great

**Comment by ralvarez76** [comment]
I ended up changing my strategy and doing more in Rhino. The modeling commands were just not robust / stable enough. When I was last looking at this it seemed to not like vey small edges when doing the patch.

On another attempt I brought in the surfaces via SAT (originally modeled in CATIA) and extracted their perimeter edges and their U / V parameter lines. I then used the Revit "form" nodes to loft surfaces our of the U / V curves. In Revit I used the perimeter curves to create a void, which then carved the edges of each surface properly. Even after all of this, a few of the surfaces would still not loft, etc.
